content/redirect.js

```diff
@@ -172,8 +194,9 @@ function add(locale, updatePairs, { fix = false } = {}) {
     );
   }
 
-  const decodedUpdatePairs = decodePairs(updatePairs);
+  // This is dangerous an we should move this burden to the caller!
```

Comment:
I *ignored* this initially because it's a mess. But we should try to introduce strict APIs from where on out URL are _decoded_ and _lowercase_. I hope this doesn't come back at us soon :/

Reply:
I don't get it. Imagine, in the future, where this PR comment isn't around. How do you connect that code comment with the `  const decodedPairs = decodePairs(pairs);` line?

Even with the PR comment I still don't understand the point of the "This is dangerous ..." comment. Sorry. Is calling `decodePairs()` dangerous??

Reply:
Maybe I'm just paranoid. We shouldn't need this here as the the `_redirects.txt` should always be decoded since we changed it. We may move to the convention that whatever calls `.add(` does so with _decoded_ pairs. Then we can remove it all together. And maybe have a warning if an internal (beginning with /) URL contains a `%`.

I would do that in a follow up?

Reply:
What I mean is that the comment is very hard to read. What's so "dangerous"?
Is it meant to be one of those `// TODO: ...` notes? 
What can go wrong if you call `decodePairs(pairs);` and what effects are there? And if it's bad one, why is it called?

I'm just pointing out that the comment isn't helping. It only makes me nervous without actually understanding the underlying **why**.

Reply:
Beyond the nit about the code comment... You mentioned that decoding should maybe only be done n the `.add()`. Perhaps I don't fully comprehend exactly what's going on but I think if someone ignores the CLI and manually clobbers in some junk into the `_redirects.txt` using the GitHub UI or emacs directly, it must get caught in CI when it eventually becomes a PR. Does that make sense?